### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by chimwemwe007**
# Blog app - Add forms

### the following has bbeen accomplished in this pull request.

- create a method called current_user in your ApplicationController to make current user data available to all controllers.
- It will return the first user from the database.
- Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
- Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
- Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+    respond_to do |format|
+      format.html do
+        if @post.save
+          redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+          # redirect_to '/users/#{@post_new.user_id}/posts', messaage: 'Post was successfully created.'
+        else
+          render :new, status: 'Error: Post was not created.'
+        end
+      end
+    end
```
Comment (jaflih) [inline app/controllers/posts_controller.rb:27]: You have a bug in your project when I create a new post

image


**Follow my others comments to correct it.**

File: db/migrate/20221025191034_create_users.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
-class CreateUser < ActiveRecord::Migration[7.0]
+class CreateUsers < ActiveRecord::Migration[7.0]
   def change
     create_table :users do |t|
       t.string :name
-      t.string :photo
-      t.string :bio
+      t.text :photo
+      t.text :bio
       t.integer :posts_counter
```
Comment (jaflih) [inline db/migrate/20221025191034_create_users.rb:7]: Please correct this line like this to remove the bug

` t.integer :posts_counter, default: 0`

File: db/migrate/20221025191732_create_posts.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
-class CreatePost < ActiveRecord::Migration[7.0]
+class CreatePosts < ActiveRecord::Migration[7.0]
   def change
     create_table :posts do |t|
       t.string :title
-      t.text :text
+      t.string :text
       t.integer :comments_counter
```
Comment (jaflih) [inline db/migrate/20221025191732_create_posts.rb:6]: Please correct this line like this to remove the bug

`t.integer :comments_counter, default: 0`

File: db/migrate/20221025191732_create_posts.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
       t.integer :likes_counter
```
Comment (jaflih) [inline db/migrate/20221025191732_create_posts.rb:7]: Please correct this line like this to remove the bug

`t.integer :likes_counter, default: 0`

**Review by jaflih — CHANGES_REQUESTED** [review]
Hi @chimwemwe007 , 

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

**Comment by chimwemwe007** [comment]
The other branch was deleted due to technical issues, and i accidentally re-submitted the deleted branch link. The other one is this link https://github.com/chimwemwe007/blog-app/pull/8 which is currently non existent, hence this link https://github.com/chimwemwe007/blog-app/pull/9. Thank you and i understand about the new pull requests.

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,40 @@
...
+# reviewer request
+ #updating 
+ # The migration
+#add_reference :posts, :author, index: true
+#add_foreign_key :posts, :users, column: :author_id
+
+# The model
+#belongs_to :author, class_name: "User"
+
+AJANLEKOKO ARIBIDESI
```
Comment (FrederickMih) [inline db/seeds.rb:44]: - [**Optional**] This uninitialized constant prevented us from running the command `rails db:seed` hence we could only create the user, post, and comment manually using the rails console. Let's kindly fix :wrench: the bug :bug: Below is the screenshot of the error on my console: 

Screenshot from 2022-11-05 16-08-40

File: app/views/posts/_add_comment.html.erb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <%= form_with url: "/users/#{@post.author_id}/posts/#{@post.id}/comments", scope: :comment do |format| %>
+    <div>
+        <%= format.label :Enter_Comment %><br>
+        <%= format.text_area :text, rows: 10, require: true %>
+        <%= format.submit "Comment" %>
+    </div>
+    <% end %>
+</div>
```
Comment (FrederickMih) [inline app/views/posts/_add_comment.html.erb:9]: - [**Optional**] With Rails, we have a unique default way of organizing our work hence, this form could have been created in the path **views/comments/new.html.erb**. This is really important, especially when scaling or debugging our work in the future.

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :heavy_check_mark: 


Hi @chimwemwe007 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




## Highlights

- Great communication with reviewers :heavy_check_mark: 
- Correct Gitflow has been followed :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Optional suggestions

- [**Optional**] The title(Fix linters and make review changes) of your PR is really poor. It basically says nothing concerning what you are implementing(Blog app - add forms)
- You were told to include all the commands necessary to run your app locally but you didn't. Would you fix it? This way your README.md file will be really interesting and descriptive.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — COMMENTED** [review]
# Hi @chimwemwe007,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Descriptive PR⭐
- [ ]  No Linters error 🤩
- [ ] Good job working on the forms💪🏻
- [ ] Good job working on the routes 🙏🏻



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**